forc-pkg/src/pkg.rs

```diff
@@ -275,28 +275,50 @@ impl BuildPlan {
     }
 
     /// Attempt to load the build plan from the `Lock`.
-    pub fn from_lock(proj_path: &Path, lock: &Lock, sway_git_tag: &str) -> Result<Self> {
-        let graph = lock.to_graph()?;
+    pub fn from_lock(
+        proj_path: &Path,
+        lock: &Lock,
+        sway_git_tag: &str,
+    ) -> Result<(Self, Vec<String>)> {
+        let mut graph = lock.to_graph()?;
+        let tmp_compilation_order = compilation_order(&graph)?;
+        let (path_map, removed_deps) =
+            graph_to_path_map(proj_path, &graph, &tmp_compilation_order, sway_git_tag)?;
+        let removed_deps_vec = removed_deps
+            .clone()
+            .into_iter()
+            .map(|dep| graph[dep].name.clone())
+            .collect();
+        // Remove the unresolveable path dependencies from the graph
+        apply_diff_after_lock(removed_deps, &mut graph, &path_map, sway_git_tag)?;
+        // Since path_map removes removed dependencies from the graph compilation order may change
         let compilation_order = compilation_order(&graph)?;
-        let path_map = graph_to_path_map(proj_path, &graph, &compilation_order, sway_git_tag)?;
-        Ok(Self {
-            graph,
-            path_map,
-            compilation_order,
-        })
+        Ok((
+            Self {
+                graph,
+                path_map,
+                compilation_order,
+            },
+            removed_deps_vec,
+        ))
     }
 
     /// Attempt to load the build plan from the `Forc.lock` file.
-    pub fn from_lock_file(lock_path: &Path, sway_git_tag: &str) -> Result<Self> {
+    pub fn from_lock_file(lock_path: &Path, sway_git_tag: &str) -> Result<(Self, Vec<String>)> {
         let proj_path = lock_path.parent().unwrap();
         let lock = Lock::from_path(lock_path)?;
         Self::from_lock(proj_path, &lock, sway_git_tag)
     }
 
     /// Ensure that the build plan is valid for the given manifest.
-    pub fn validate(&self, manifest: &Manifest, sway_git_tag: &str) -> Result<PkgDiff> {
+    pub fn validate(
+        &self,
+        removed: Vec<String>,
+        manifest: &Manifest,
+        sway_git_tag: &str,
+    ) -> Result<PkgDiff> {
```

Comment:
Might be useful for this method's docs to mention that the `removed` packages represent the set of packages already removed from the graph due to invalid paths.